### PR TITLE
Fixed issue #17950: is_int(false) returns true

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: false and '' fixed

### CODE REVIEW

**Comment by Shnoulle** [comment]
Still issue with `.00`

**Comment by Shnoulle** [comment]
@c-schmitz what about .000 : must send false always ?

`.001` send false, then i think `.000` must send false too

**Comment by Shnoulle** [comment]
For 3.X : https://github.com/LimeSurvey/LimeSurvey/pull/2431

**Review by gabrieljenik — APPROVED** [review]
Code structure looks OK. Haven't tested it

File: application/helpers/expressions/em_core_helper.php [hunk 1]
```diff
@@ -2941,18 +2941,23 @@ function exprmgr_if($testDone, $iftrue, $iffalse = '')
...
  * @link http://php.net/is_int#82857
  */
 function exprmgr_int($arg)
 {
     if (strpos($arg, ".")) {
-        $arg = preg_replace("/\.$/", "", rtrim(strval($arg), "0")); // DECIMAL from SQL return always .00000000, the remove all 0 and one . , see #09550
+        // DECIMAL from SQL return always .00000000, the remove all 0 and one . , see #09550
+        $arg = preg_replace("/\.$/", "", rtrim(strval($arg), "0"));
     }
-    return (preg_match("/^-?[0-9]*$/", $arg)); // Allow 000 for value, @link https://bugs.limesurvey.org/view.php?id=9550 DECIMAL sql type.
```
[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/helpers/expressions/em_core_helper.php:2954]: the link is here https://github.com/LimeSurvey/LimeSurvey/pull/2432/files#diff-a974f1aca3b64170e04f9ac415aadb10d9108696f0ae4ce07caee89d8a1447d3R2953
But you're right, it's not related to 000 here, maybe add the information …

[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/helpers/expressions/em_core_helper.php:2954]: It's done line 2956

**Comment by olleharstedt** [comment]
Can we add a unit test for this?

**Comment by Shnoulle** [comment]
> Can we add a unit test for this? Without scripting the browser, just test the PHP function.

It's great if we have https://github.com/LimeSurvey/LimeSurvey/blob/master/tests/unit/helpers/ExpressionCoreHelperTest.php ready …

**Comment by Shnoulle** [comment]
[Shit issue when https://github.com/LimeSurvey/LimeSurvey/pull/2642merging ?](https://github.com/LimeSurvey/LimeSurvey/pull/2642)

**Comment by Shnoulle** [comment]
https://github.com/LimeSurvey/LimeSurvey/pull/2642

**Pull request closed without merging**